system_tests/datastore.py

```diff
@@ -432,6 +432,43 @@ def test_transaction(self):
         self.case_entities_to_delete.append(retrieved_entity)
         self.assertEqual(retrieved_entity, entity)
 
+    def test_transaction_via_explicit_begin_get_commit(self):
```

Comment:
Should we make this test more generic and less tailored to the specific use case that was reported in the issue? I'm good either way just something I was thinking about.